### PR TITLE
Replace dependencies to improve bundle size

### PR DESCRIPTION
**Pull request by gillchristian**
Replaced dependencies
- `qs` with [query-string](https://github.com/sindresorhus/query-string): no more nested object, so those tests were removed.
- `lodash/merge` with a small util that used `Object.assign` and specially handles `headers` since is the only nested thing in our config. This one needs to be improved a bit, but the solution works (at least tests pass 🤐)

Before (master)

Screenshot 2020-03-17 at 23 59 14

After

Screenshot 2020-03-17 at 23 58 42

### CODE REVIEW

File: test/lib.url.spec.ts [hunk 1]
```diff
@@ -3,46 +3,39 @@
...
+      const url = 'https://www.foo.com/bar';
 
       expect(actual).toBe(expectted);
     });
 
     it('when params are an empty object it returns the same url', () => {
       const url = 'https://www.foo.com/bar';
-      const params = {}
-      const actual = concatParams(url, {})
+      const params = {};
```
Comment (gillchristian) [inline test/lib.url.spec.ts]: We need to fix (or add) prettier hooks. My editor formatted this file, so now we have unnecessary diff 🤷‍♂

File: src/utils.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+type WithHeaders = { headers: unknown };
+
+// TODO: do not use `any`
+export const merge = (...objs: any[]) => {
+  const headers = objs
+    .filter((obj): obj is WithHeaders =>
+      typeof obj === 'object' && obj ? Boolean(obj.headers) : false,
```
Comment (ndelvalle) [inline src/utils.ts]: I think if `typeof obj === 'object'` statement is true we do not need the `&& obj` because it would be redundant.

Maybe `typeof obj === 'object' && !!obj.headers`?

File: test/lib.url.spec.ts [hunk 3]
```diff
@@ -3,46 +3,39 @@
...
-          }
-        }
-      };
-      const actual = concatParams(url, params)
-      const expectted = 'https://www.foo.com/bar?foo%5Bbar%5D%5Bbaz%5D=foobarbaz'
+      const url = 'https://www.foo.com/bar';
+      const params = { bar: 'bar', foo: 'foo' };
+      const actual = concatParams(url, params);
+      const expectted = 'https://www.foo.com/bar?bar=bar&foo=foo';
+      const url = 'https://www.foo.com/bar';
```
Comment (ndelvalle) [inline test/lib.url.spec.ts]: This `url` is already defined at line 9 😇

Comment (gillchristian) [reply test/lib.url.spec.ts]:
```suggestion
```

**Comment by ndelvalle** [comment]
This bundle size improvement is great! 🙌🏻

File: src/utils.ts (hunk 2 above)
Comment (gillchristian) [reply src/utils.ts]: `typeof null` is `object` 😕

Comment (gillchristian) [reply src/utils.ts]:
```suggestion
      typeof obj === 'object' && obj ? !!obj.headers : false,
```

I normally do casting with `Boolean` than `!!`, personal preference. But `!!` is shorter so build will be smaller :trollface:

Comment (ndelvalle) [reply src/utils.ts]: Yeah, I prefer `!!` because of performance concerns 😅. Jokes aside I do not mind the `Boolean` I just used `!!` in the example because I am used to that.
